src/main/java/com/fasterxml/jackson/core/JsonGenerator.java

```diff
@@ -933,6 +933,34 @@ public void writeArray(double[] array, int offset, int length) throws IOExceptio
     public abstract void writeString(String text) throws IOException;
 
     /**
+     * Current {@link #writeString(String)} method requires text value to be passed in its entirety.
+     * This may not work well with larger payloads; so we should an alternative
+     * method where one can define source for text, used for incremental/streaming read+write.
+     * @param reader
+     * @param charLength if this parameter is -1 that means we have a unknown
+     *                   size of char array. But all generators are required to support it.
+     *                   If length is indicated, it is an error for Reader to provide less data
+     *                   (more is ok; generator just will not read anything beyond indicated amount).
+     * @throws IOException
+     */
+    public void writeString(Reader reader, int charLength) throws IOException {
+        if (reader == null) {
+            return;
+        }
+
+        if (charLength == -1) {
+            while (reader.read() != -1) ;
```

Comment:
Uh, what? If not known, quietly swallow content?!?!? And do that as inefficiently as possible, char by char, instead of skipping?